### PR TITLE
Prevent instrumentation of com.ibm.jit.JITHelpers

### PR DESCRIPTION
**Pull request by theresa-m**
Issue: #2768

Signed-off-by: Theresa Mammarella <Theresa.T.Mammarella@ibm.com>

### CODE REVIEW

File: runtime/util/hshelp.c [hunk 1]
```diff
@@ -3229,6 +3230,8 @@ classIsModifiable(J9JavaVM * vm, J9Class * clazz)
 		rc = JNI_FALSE;
 	} else if ((J2SE_VERSION(vm) >= J2SE_19) && J9_ARE_ALL_BITS_SET(clazz->classFlags, J9ClassIsAnonymous)) {
 		rc = JNI_FALSE;
+	} else if (strcmp(J9UTF8_DATA(J9ROMCLASS_CLASSNAME(clazz->romClass)), "com/ibm/jit/JITHelpers") == 0) {
```
Comment (DanHeidinga) [inline runtime/util/hshelp.c]: `J9UTF8_DATA(J9ROMCLASS_CLASSNAME(clazz->romClass))` isn't guaranteed to be null terminated.

We need to check the length matches the length of `com/ibm/jit/JITHelpers` before doign the strcmp

Comment (theresa-m) [reply runtime/util/hshelp.c]: my fix checks that the class name length is either 22 (com/ibm/jit/JITHelper without terminator) or 23 (com/ibm/jit/JITHelper with terminator) before comparing characters

File: runtime/util/hshelp.c [hunk 2]
```diff
@@ -3229,6 +3232,9 @@ classIsModifiable(J9JavaVM * vm, J9Class * clazz)
 		rc = JNI_FALSE;
 	} else if ((J2SE_VERSION(vm) >= J2SE_19) && J9_ARE_ALL_BITS_SET(clazz->classFlags, J9ClassIsAnonymous)) {
 		rc = JNI_FALSE;
+	} else if (((classNameLength == 22) || (classNameLength == 23))
```
Comment (DanHeidinga) [inline runtime/util/hshelp.c]: length won't include the null terminator.  

Can this be written with an `else ` clause that has the name based matching in it?

ie:
```
} else {
 	/* Handle name based matches */
 	U_8* className = J9UTF8_DATA(J9ROMCLASS_CLASSNAME(clazz->romClass));
 	U_8 classNameLength = J9UTF8_LENGTH(className);
 	switch(classNameLength) {
 	case 22:
 	 	if (strncmp((char*)className, "com/ibm/jit/JITHelpers", 22) == 0) {
 	 	 	 rc = JNI_FALSE;
 	 	 }
 	 	break;
 	}
```

Comment (theresa-m) [reply runtime/util/hshelp.c]: updated

File: runtime/util/hshelp.c [hunk 3]
```diff
@@ -3229,6 +3230,18 @@ classIsModifiable(J9JavaVM * vm, J9Class * clazz)
 		rc = JNI_FALSE;
 	} else if ((J2SE_VERSION(vm) >= J2SE_19) && J9_ARE_ALL_BITS_SET(clazz->classFlags, J9ClassIsAnonymous)) {
 		rc = JNI_FALSE;
+	} else {
+		/* Handle name based matches */
+		U_8* className = J9UTF8_DATA(J9ROMCLASS_CLASSNAME(clazz->romClass));
+		U_8 classNameLength = J9UTF8_LENGTH(className);
+
+		switch(classNameLength) {
+			case 22:
```
Comment (DanHeidinga) [inline runtime/util/hshelp.c]: Nitpick: Our typical coding convention is to line the case up with the switch and to always have a break statement

```
switch(classNameLength) {
case 22:
	if (strncmp((char*)className, "com/ibm/jit/JITHelpers", 22) == 0) {
	......
	break;
default:
       rc = JNI_TRUE;
}
```

File: runtime/util/hshelp.c [hunk 4]
```diff
@@ -3229,6 +3230,20 @@ classIsModifiable(J9JavaVM * vm, J9Class * clazz)
 		rc = JNI_FALSE;
 	} else if ((J2SE_VERSION(vm) >= J2SE_19) && J9_ARE_ALL_BITS_SET(clazz->classFlags, J9ClassIsAnonymous)) {
 		rc = JNI_FALSE;
+	} else {
+		/* Handle name based matches */
+		U_8* className = J9UTF8_DATA(J9ROMCLASS_CLASSNAME(clazz->romClass));
+		U_8 classNameLength = J9UTF8_LENGTH(className);
```
Comment (DanHeidinga) [inline runtime/util/hshelp.c]: This should be `J9UTF8_LENGTH(J9ROMCLASS_CLASSNAME(clazz->romClass));`

The length isn't available from the data.

Comment (theresa-m) [reply runtime/util/hshelp.c]: updated

**Comment by theresa-m** [comment]
related issue has been closed

**Pull request closed without merging**